src/prpy/util.py

```diff
@@ -749,3 +749,168 @@ def IsTimedTrajectory(trajectory):
     """
     cspec = trajectory.GetConfigurationSpecification()
     return cspec.ExtractDeltaTime(trajectory.GetWaypoint(0)) is not None
+
+
+def JointStatesFromTraj(robot, traj, times):
+    """
+    Helper function to extract the joint position, velocity and acceleration
+    from an OpenRAVE trajectory.
+    @param robot The OpenRAVE robot
+    @param traj An OpenRAVE trajectory
+    @param times List of times in seconds
+    @return pva_list List of list of position, velocity and acceleration
+                     at specified times. Inserts 'None' for unavailable fields
+    """
+
+    duration = traj.GetDuration()
+
+    if any(times > duration):
+        raise ValueError('Input times {0:} exceed duration {1:.2f}'
+                         .format(times, duration))
+
+    cspec = traj.GetConfigurationSpecification()
+    num_dofs = robot.GetDOF()
+    dof_indices = range(num_dofs)
+
+    pva_list = []
+    for t in times:
+        pva = []
+        trajdata = traj.Sample(t)
+        pva.append(cspec.ExtractJointValues(trajdata, robot, dof_indices, 0))
+        pva.append(cspec.ExtractJointValues(trajdata, robot, dof_indices, 1))
+        pva.append(cspec.ExtractJointValues(trajdata, robot, dof_indices, 2))
```

Comment:
This is an excellent idea. But, I worry that `JointState` now might have variable size, depending on the derivatives that are requested. I fear this will lead to the same discomfort we have with _active_ joints. 

I feel like it's more natural if something not requested is `None` rather than be arbitrarily shortened.